consensus/beacon/consensus.go

```diff
@@ -372,6 +373,14 @@ func (beacon *Beacon) FinalizeAndAssemble(chain consensus.ChainHeaderReader, hea
 			return nil, errors.New("withdrawals set before Shanghai activation")
 		}
 	}
+	if chain.Config().IsCancun(header.Number, header.Time) {
+		var blobs int
+		for _, tx := range txs {
+			blobs += len(tx.BlobHashes())
+		}
+		dataGasUsed := uint64(blobs * params.BlobTxDataGasPerBlob)
+		header.DataGasUsed = &dataGasUsed
+	}
```

Comment:
Is this really the best place to update the header? It kind of seems like this belongs into the state transition, where we usually track the block gas?